src/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,440 @@
+'use strict';
+
+const Base = require('./Base');
+const { Error } = require('../errors');
+const {
+  GuildScheduledEventEntityTypes,
+  GuildScheduledEventStatuses,
+  GuildScheduledEventPrivacyLevels,
+  Endpoints,
+} = require('../util/Constants');
+const SnowflakeUtil = require('../util/SnowflakeUtil');
+
+/**
+ * @type {WeakSet<GuildScheduledEvent>}
+ * @private
+ * @internal
+ */
+const deletedGuildScheduledEvents = new WeakSet();
+
+/**
+ * Represents a scheduled event in a {@link Guild}.
+ * @extends {Base}
+ */
+class GuildScheduledEvent extends Base {
+  constructor(client, data) {
+    super(client);
+
+    /**
+     * The id of the guild scheduled event
+     * @type {Snowflake}
+     */
+    this.id = data.id;
+
+    /**
+     * The id of the guild this guild scheduled event belongs to
+     * @type {Snowflake}
+     */
+    this.guildId = data.guild_id;
+
+    this._patch(data);
+  }
+
+  _patch(data) {
+    if ('channel_id' in data) {
+      /**
+       * The channel id in which the scheduled event will be hosted, or `null` if entity type is `EXTERNAL`
+       * @type {?Snowflake}
+       */
+      this.channelId = data.channel_id;
+    } else {
+      this.channelId ??= null;
+    }
+
+    if ('creator_id' in data) {
+      /**
+       * The id of the user that created this guild scheduled event
+       * @type {?Snowflake}
+       */
+      this.creatorId = data.creator_id;
+    } else {
+      this.creatorId ??= null;
+    }
+
+    /**
+     * The name of the guild scheduled event
+     * @type {string}
+     */
+    this.name = data.name;
+
+    if ('description' in data) {
+      /**
+       * The description of the guild scheduled event
+       * @type {?string}
+       */
+      this.description = data.description;
+    } else {
+      this.description ??= null;
+    }
+
+    /**
+     * The timestamp the guild scheduled event will start at
+     * <info>This can be potentially `null` only when it's an {@link AuditLogEntryTarget}</info>
+     * @type {?number}
+     */
+    this.scheduledStartTime = data.scheduled_start_time ? new Date(data.scheduled_start_time).getTime() : null;
```

Comment:
Should we be consistent and suffix them in `Timestamp`?

Reply:
Makes sense but what about keeping consistency with API naming?

Reply:
`scheduledStartTimeTimestamp` storing a timestamp and `scheduledStartTime` being a getter that returns a `Date` from it makes sense to me.

Reply:
date getters have `At` at the end, we would be breaking one consistency to keep another.
<sub>ℹ️ Current names: `scheduledStartAt` & `scheduledEndAt`</sub>

Reply:
Or neither: `startsTimestamp` and `startsAt`? More consistent with what we have (similar to `{joined,created}{At,Timestamp}`), but differs a lot from Discord's naming?

Reply:
If we are diverging from Discord's naming this time then I would suggest: `scheduledStartTimestamp` & keep the `scheduledStartAt`.

Reply:
I personally agree with @iShibi here